cli/src/main/java/org/openjdk/skara/cli/GitPr.java

```diff
@@ -519,7 +519,11 @@ public static void main(String[] args) throws IOException, InterruptedException
 
                 System.exit(0);
             }
-            var currentBranch = repo.currentBranch();
+            var currentBranch = repo.currentBranch().orElseGet(() -> {
+                    System.err.println("error: the repository is in a detached HEAD state");
```

Comment:
```suggestion
                    System.err.println("error: the repository is in a detached HEAD state. check out a branch to continue.. or something");
```

Reply:
I thought about this as well, but if you see the "detached HEAD" warning when doing e.g. `git checkout HEAD~` there is just no way we are gonna be able to write all that :smiley:

So I figured that the next best thing was to write something concise that is very easy to search for. If the user does not know what "detached HEAD state" means then he/she will immediately search for it and will most likely stumble upon [this explanation](https://www.git-tower.com/learn/git/faq/detached-head-when-checkout-commit) which is the first hit on both DuckDuckGo and Google for me.

I think that is the best alternative here, what do you say?

Reply:
Sounds good!